### PR TITLE
Component

### PR DESCRIPTION
**Pull request by saeedark**
1. Add hierarchical platform for networkobjects.
2. Allows synapses to be created without src and dst.

### CODE REVIEW

File: pymonntorch/NetworkCore/Network.py [hunk 1]
```diff
@@ -116,7 +122,7 @@ def clear_recorder(self, keys=None):
     def __repr__(self):
         neuron_count = torch.sum(torch.tensor([ng.size for ng in self.NeuronGroups]))
         synapse_count = torch.sum(
-            torch.tensor([sg.src.size * sg.dst.size for sg in self.SynapseGroups])
+            torch.tensor([sg.src.size * sg.dst.size for sg in self.SynapseGroups if (sg.src is not None and sg.dst is not None)])
```
Comment (realamirhe) [inline pymonntorch/NetworkCore/Network.py:125]: what is the purpose of if statement

File: pymonntorch/NetworkCore/Network.py [hunk 2]
```diff
@@ -146,6 +152,10 @@ def __repr__(self):
                 result += str(sg) + "\r\n"
             used_tags.append(tags)
 
+        for struc in self.Structures:
+            if struc not in self.NeuronGroups and struc not in self.SynapseGroups:
+                result += str(struc) + "\r\n"
```
Comment (realamirhe) [inline pymonntorch/NetworkCore/Network.py:157]: is "carriage return" needed 😅 ?

File: pymonntorch/NetworkCore/NeuronGroup.py [hunk 3]
```diff
@@ -22,7 +22,7 @@ class NeuronGroup(NetworkObject):
         id (torch.Tensor): The integer id of the population.
     """
 
-    def __init__(self, size, behavior, net, tag=None):
+    def __init__(self, size, net, behavior=None, tag=None):
```
Comment (realamirhe) [inline pymonntorch/NetworkCore/NeuronGroup.py:25]: could we set a convention that net/network is always the first parameter

File: pymonntorch/NetworkCore/Network.py (hunk 1 above)
Comment (saeedark) [reply pymonntorch/NetworkCore/Network.py:125]: A SynapseGroup can be created without a src or dst population therefore calculating the number of synapses is futile.

File: pymonntorch/NetworkCore/NeuronGroup.py (hunk 3 above)
Comment (saeedark) [reply pymonntorch/NetworkCore/NeuronGroup.py:25]: There will be another PR fixing these issues plus type hints for arguments.

File: pymonntorch/NetworkCore/Network.py (hunk 2 above)
Comment (saeedark) [reply pymonntorch/NetworkCore/Network.py:157]: Apparently not, Old convention, gonna fix it in another PR.

**Review by realamirhe — APPROVED** [review]

**Pull request merged**